src/Tinter.js

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2015 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+var dis = require('./dispatcher');
+
+var registered = false;
+if (!registered) {
+    dis.register(_onAction);
+}
+
+var keyRgb = [
+    "rgb(118, 207, 166)",
+    "rgb(234, 245, 240)",
+    "rgba(118, 207, 166, 0.2)",
+];
+
+// Some algebra workings for calculating the tint % of Vector Green & Light Green
+// x * 118 + (1 - x) * 255 = 234
+// x * 118 + 255 - 255 * x = 234
+// x * 118 - x * 255 = 234 - 255
+// (255 - 118) x = 255 - 234
+// x = (255 - 234) / (255 - 118) = 0.16 
+
+var keyHex = [
+    "#76CFA6",
+    "#EAF5F0",
+    "#D3EFE1",
+];
+
+// cache of our replacement colours
+// defaults to our keys.
+var colors = [
+    keyHex[0],
+    keyHex[1],
+    keyHex[2],
+];
+
+var cssFixups = [
+    // {
+    //     style: a style object that should be fixed up taken from a stylesheet
+    //     attr: name of the attribute to be clobbered, e.g. 'color'
+    //     index: ordinal of primary, secondary or tertiary
+    // }
+];
+
+// CSS attributes to be fixed up
+var cssAttrs = [
+    "color",
+    "backgroundColor",
+    "borderColor",
+];
+
+var svgFixups = [
+    // {
+    //     node: a SVG node that needs to be fixed up
+    //     attr: name of the attribute to be clobbered, e.g. 'fill'
+    //     index: ordinal of primary, secondary
+    // }
+];
```

Comment:
I think I finally grok how and why you're doing this. I don't like it because it is leaking DOM nodes all over the place because `Tinter` is a singleton when really the things it is manipulating are UI elements. Either strip this class to be pure-functional, or move it all to a separate UI component (the one you're writing). This means that you would have something like:

```javascript
React.createClass({

  _fixup: function(event) {
    // do fixup logic that Tinter.js is doing
  },

  render: function() {
    return <object onload={this._fixup} className="mx_Svg" type="image/svg+xml" data="img/leave.svg" />
});
```

This is much nicer since this will drop out of scope when the component is unmounted, stopping any leaks. My next questions would be:
 - Do you need `mx_Svg` anymore given you no longer need to do a full-scan of `document`?
 - You still need to retroactively apply tints to mounted svg components when people type `/tint`. This would be something that would be useful to be done via the `dispatcher` so you don't need to maintain a list of currently mounted svg components. So you'd have:
    ```javascript
    onAction: function(payload) {
      if (payload.action !== "tint_change") { return; }
      // apply new tint values from payload
    }
    ```